felix/fv/wireguard_test.go

```diff
@@ -86,24 +86,26 @@ var _ = infrastructure.DatastoreDescribe("_BPF-SAFE_ WireGuard-Supported", []api
 	)
 
 	type testConf struct {
-		WireguardEnabledV4 bool
-		WireguardEnabledV6 bool
+		WireguardEnabledV4        bool
+		WireguardEnabledV6        bool
+		WireguardThreadingEnabled bool
 	}
 	for _, testConfig := range []testConf{
-		{true, false},
-		{false, true},
-		{true, true},
+		{true, false, false},
+		{false, true, true},
+		{true, true, false},
```

Comment:
One extra change I had in mind, is to revert the testConf struct to its original format. Then in the test, update the Felix threading value, and then use `Eventually` to wait for the change in the corresponding sysfs file.

One main reason is that in the current tests, only the case with wireguard v4 is actually tested. There is no test case for wireguard v6.

Reply:
Got it. That looks cleaner as well.